dll/win32/shell32/shellmenu/CStartMenu.cpp

```diff
@@ -249,6 +308,16 @@ class CShellMenuCallback :
             ShellExecuteW(NULL, NULL, L"explorer.exe", L"::{20D04FE0-3AEA-1069-A2D8-08002B30309D}\\::{21EC2020-3AEA-1069-A2DD-08002B30309D}\\::{7007ACC7-3202-11D1-AAD2-00805FC1270E}", NULL, SW_SHOWNORMAL);
         else if (psmd->uId == IDM_PRINTERSANDFAXES)
             ShellExecuteW(NULL, NULL, L"explorer.exe", L"::{20D04FE0-3AEA-1069-A2D8-08002B30309D}\\::{21EC2020-3AEA-1069-A2DD-08002B30309D}\\::{2227A280-3AEA-1069-A2DE-08002B30309D}", NULL, SW_SHOWNORMAL);
+        else if (psmd->uId == IDM_MYDOCUMENTS)
+        {
+            SHGetSpecialFolderPathW(NULL, szPath, CSIDL_PERSONAL, FALSE);
+            ShellExecuteW(NULL, NULL, szPath, NULL, NULL, SW_SHOWNORMAL);
+        }
+        else if (psmd->uId == IDM_MYPICTURES)
+        {
+            SHGetSpecialFolderPathW(NULL, szPath, CSIDL_MYPICTURES, FALSE);
```

Comment:
take care of SHGetSpecialFolderPathW return value before calling shellexecute

Reply:
Added check. OK.